### PR TITLE
Ac/refac/chang all comps to func

### PR DESCRIPTION
**Pull request by aviadch**
This is a very big branch (not in the amount but in the mindset)

All components are now functions.
also I change the context api to work fine with these function components.

code look much better now!

### CODE REVIEW

File: frontend/src/components/Player.js [hunk 1]
```diff
@@ -1,19 +1,18 @@
...
-          src={require("./player1.jpg")}
-          alt={this.props.name}
-          height="100"
-          width="100"
-        />
-      </span>
-    );
-  }
-}
+const Player = (props) => {
```
Comment (AdiYElisha) [inline frontend/src/components/Player.js:3]: Player sounds like an entity, maybe chaaaaaange to playerCard?

File: frontend/src/infra/MyProvider.js [hunk 2]
```diff
@@ -30,45 +21,57 @@ class MyProvider extends Component {
...
-  };
+  });
+
+  socket.on("NewPlayer", (data) => {
+    console.log("New Player from socket");
+    setState({ ...state, playerList: data.roomPlayers });
+  });
 
-  onNewGame = () => {
+  const onNewGame = () => {
```
Comment (AdiYElisha) [inline frontend/src/infra/MyProvider.js:31]: new room or new game?

File: frontend/src/infra/MyProvider.js [hunk 3]
```diff
@@ -30,45 +21,57 @@ class MyProvider extends Component {
...
           roomCreationDate: creationDate,
         });
+        console.log(state);
       });
+
+    console.log(state);
   };
-  joinGame = (roomId) => {
+
+  const joinGame = (roomId) => {
```
Comment (AdiYElisha) [inline frontend/src/infra/MyProvider.js:54]: join room? join game?

File: frontend/src/infra/MyProvider.js [hunk 4]
```diff
@@ -30,45 +21,57 @@ class MyProvider extends Component {
...
+    setState({
+      ...state,
       phase: PHASE.WAITING_ROOM,
       currentRoom: roomId,
       playerId: playerId,
     });
   };
-
-  onNameEntered = (name) => {
+  const onNameEntered = (name) => {
```
Comment (AdiYElisha) [inline frontend/src/infra/MyProvider.js:63]: NameEntered? or onUserEntered

Comment (AdiYElisha) [reply frontend/src/infra/MyProvider.js:63]: or onNameSet

File: frontend/src/screens/WaitingRoom.js [hunk 5]
```diff
@@ -1,51 +1,30 @@
...
-                {!context.state.isUserEnteredName ? (
-                  <>
-                    <InputName />
-                  </>
-                ) : (
-                  <></>
-                )}
+  return (
+    <div className="waiting-room">
+      <div>
```
Comment (AdiYElisha) [inline frontend/src/screens/WaitingRoom.js:13]: I don't think this div is needed

File: frontend/src/screens/WaitingRoom.js [hunk 6]
```diff
@@ -1,51 +1,30 @@
...
+        <Typography variant="h1" component="h2" gutterBottom>
+          Waiting Room - RoomId: {context.state.currentRoom}
+        </Typography>
+        {!context.state.isUserEnteredName ? <EnterName /> : null}
```
Comment (AdiYElisha) [inline frontend/src/screens/WaitingRoom.js:17]: `!context.state.isUserEnteredName && <EnterName /> ` is not what you need?

File: frontend/src/screens/WelcomeScreen.js [hunk 7]
```diff
@@ -1,63 +1,50 @@
...
+const WelcomeScreen = () => {
+  const context = useContext(MyContext);
+  console.log(context.state.phase);
+  const [roomToJoin, setRoom] = useState("");
+
+  const onChange = (event) => {
     const value = event.target.value;
-    console.log(this.state.roomToJoin);
-    this.setState({ roomToJoin: value });
+    setRoom(value);
```
Comment (AdiYElisha) [inline frontend/src/screens/WelcomeScreen.js:15]: what does that mean?

File: frontend/src/screens/WelcomeScreen.js [hunk 8]
```diff
@@ -1,63 +1,50 @@
...
-      roomToJoin: ""
-    };
-  }
-  onChange = event => {
+const WelcomeScreen = () => {
+  const context = useContext(MyContext);
+  console.log(context.state.phase);
+  const [roomToJoin, setRoom] = useState("");
+
+  const onChange = (event) => {
```
Comment (AdiYElisha) [inline frontend/src/screens/WelcomeScreen.js:13]: on what change?

File: frontend/src/screens/WelcomeScreen.js [hunk 9]
```diff
@@ -1,63 +1,50 @@
...
-                    Join Game
-                  </Button>
-                </div>
-              </>
-            )}
-          </MyContext.Consumer>
+  return (
+    <div className="welcome-screen">
+      <Typography variant="h1" component="h2">
+        Hey - Welcome to Mafia Game
```
Comment (AdiYElisha) [inline frontend/src/screens/WelcomeScreen.js:22]: remove the hey

File: frontend/src/screens/WelcomeScreen.js [hunk 10]
```diff
@@ -1,63 +1,50 @@
...
-    super();
-    this.state = {
-      roomToJoin: ""
-    };
-  }
-  onChange = event => {
+const WelcomeScreen = () => {
+  const context = useContext(MyContext);
+  console.log(context.state.phase);
+  const [roomToJoin, setRoom] = useState("");
```
Comment (AdiYElisha) [inline frontend/src/screens/WelcomeScreen.js:11]: roomToJoin - what is room? roomIdToJoin maybe?

File: frontend/src/screens/WaitingRoom.js (hunk 6 above)
Comment (aviadch) [reply frontend/src/screens/WaitingRoom.js]: Great call
https://github.com/aviadch/Mafia/pull/7/commits/4cf0dab567dc11c0a4823e274851e3c481a18423

File: frontend/src/screens/WelcomeScreen.js (hunk 7 above)
Comment (aviadch) [reply frontend/src/screens/WelcomeScreen.js:15]: it more easy to debug that why but i eraesed it

File: frontend/src/screens/WelcomeScreen.js (hunk 9 above)
Comment (aviadch) [reply frontend/src/screens/WelcomeScreen.js:22]: name changed

File: frontend/src/screens/WelcomeScreen.js (hunk 10 above)
Comment (aviadch) [reply frontend/src/screens/WelcomeScreen.js:11]: changed that

File: backend/api_server.js [hunk 11]
```diff
@@ -1,10 +1,12 @@
 // import express (after npm install express)
 const express = require("express");
 const bodyParser = require("body-parser");
+const cors = require("cors");
 
 // create new express app and save it as "app"
 const app = express();
 app.use(bodyParser.json());
+app.use(cors());
```
[the remark this answers is not in the document]
Comment (aviadch) [reply backend/api_server.js:9]: same origing policy will block any request that not coming from localhost:8080.

I fixed it

File: frontend/src/MafiaApp.js [hunk 12]
```diff
@@ -0,0 +1,33 @@
...
+        break;
+
+      default:
+        screen = null;
+    }
+    return screen;
+  };
+
+  let screen = screenChooser(context.state.phase);
+  return <div className="MafiaApp">{screen}</div>;
```
[the remark this answers is not in the document]
Comment (aviadch) [reply frontend/src/MafiaApp.js:30]: changed

File: frontend/src/infra/MyProvider.js (hunk 2 above)
Comment (aviadch) [reply frontend/src/infra/MyProvider.js:20]: changed

File: frontend/src/infra/MyProvider.js (hunk 3 above)
Comment (aviadch) [reply frontend/src/infra/MyProvider.js:49]: join room, fixed

File: frontend/src/screens/WaitingRoom.js (hunk 5 above)
Comment (aviadch) [reply frontend/src/screens/WaitingRoom.js:13]: and you right.. fixed

File: frontend/src/infra/MyProvider.js (hunk 4 above)
Comment (aviadch) [reply frontend/src/infra/MyProvider.js:58]: onNameSet, correct no

File: frontend/src/infra/MyProvider.js [hunk 13]
```diff
@@ -30,8 +25,14 @@ const MyProvider = (props) => {
     axios
       .post(`${SERVER_ADDRESS}:${SERVER_PORT}/room/create`, req)
       .then((res) => {
-        const { roomID, ...creationDate } = res.data;
+        const { roomID, creationDate, socketPort } = res.data;
```
Comment (AdiYElisha) [inline frontend/src/infra/MyProvider.js:28]: socketPort shouldn't be a const?

File: frontend/src/infra/MyProvider.js [hunk 14]
```diff
@@ -30,8 +25,14 @@ const MyProvider = (props) => {
...
         console.log(`roomID:${roomID}`);
+        createSocketAndListen(socketPort, "NewPlayer", (data) => {
```
Comment (AdiYElisha) [inline frontend/src/infra/MyProvider.js:30]: maybe newPlayerJoined?

File: frontend/src/screens/WelcomeScreen.js [hunk 15]
```diff
@@ -8,12 +8,11 @@ import TextField from "@material-ui/core/TextField";
...
   const context = useContext(MyContext);
   console.log(context.state.phase);
-  const [roomToJoin, setRoom] = useState("");
+  const [roomIDToJoin, setRoom] = useState("");
 
-  const onChange = (event) => {
-    const value = event.target.value;
-    setRoom(value);
-    console.log(roomToJoin);
+  const onRoomInputChange = (event) => {
```
Comment (AdiYElisha) [inline frontend/src/screens/WelcomeScreen.js:13]: onRoomInputChange? what is room?

File: frontend/src/screens/WelcomeScreen.js [hunk 16]
```diff
@@ -8,12 +8,11 @@ import TextField from "@material-ui/core/TextField";
...
+    setRoom(event.target.value);
```
Comment (AdiYElisha) [inline frontend/src/screens/WelcomeScreen.js:14]: what is setRoom

File: backend/api_server.js (hunk 11 above)
[the remark this answers is not in the document]
Comment (aviadch) [reply backend/api_server.js]: cors will block anything not coming as a request from localhost:3000 for now.

File: frontend/src/components/InputName.js [hunk 17]
```diff
@@ -1,45 +1,33 @@
-import React, { Component } from "react";
+import React, { useState, useContext } from "react";
```
[the remark this answers is not in the document]
Comment (aviadch) [reply frontend/src/components/InputName.js]: 1ca6af0

File: frontend/src/components/Player.js (hunk 1 above)
Comment (aviadch) [reply frontend/src/components/Player.js]: bd0d2c69a06b6a971340be0b04c5d1c23b69cc98

File: frontend/src/infra/MyProvider.js (hunk 13 above)
Comment (aviadch) [reply frontend/src/infra/MyProvider.js]: Outdated

File: frontend/src/infra/MyProvider.js (hunk 14 above)
Comment (aviadch) [reply frontend/src/infra/MyProvider.js]: 0e78e3406952d13a7371f3ab0ade64af33ee4076

File: frontend/src/screens/WelcomeScreen.js (hunk 15 above)
Comment (aviadch) [reply frontend/src/screens/WelcomeScreen.js:10]: You confused from room/game naming.
It has changed. now everything is room

File: frontend/src/screens/WelcomeScreen.js (hunk 16 above)
Comment (aviadch) [reply frontend/src/screens/WelcomeScreen.js:11]: its a method we got from useState hook before, its fine. we edit the room before we send it

**Pull request merged**